book/invalidation.md

```diff
@@ -29,26 +29,15 @@ a few frames' delay is distracting. But editing changes the HTML tree
 and therefore the layout tree. Rebuilding the layout tree from
 scratch, which our browser currently does, can be very slow on
 complex pages. Try, for example, loading [this
-chapter](invalidation.md) in our browser and typing into this
-input box:
+chapter](https://browser.engineering/invalidation.html) in our browser and typing into the input box
+that appears after this paragraph...You'll find that it is *much* too
+slow---1.7 seconds just in `render` (see Figure 1)!
+^[Trace [here](https://browser.engineering/examples/example16-input-no-optimizations.trace).]
```

Comment:
Can you confirm that there's no space between the `!` and the footnote in the PDF build?

Reply:
It was not- fixed just now.